python/pyspark/sql/__init__.py

```diff
@@ -41,6 +41,13 @@
 """
 from __future__ import absolute_import
 
+
+def since(version):
```

Comment:
Quick question: Should this be in the root `pyspark` package instead of SQL?  We might want to also use this for core APIs as well.